app/assets/javascripts/components/buttons/form-button-component.js

```diff
@@ -0,0 +1,47 @@
+ManageIQ.angular.app.component('formButtonComponent', {
+  bindings: {
+    angularForm: '=',
+    modelCopy: '=',
+    model: '=',
+    redirectUrl: '=?',
+    newRecord: '<',
+    entity: '=?',
+    entityName: '=?',
+    saveClicked: '&?',
+    addClicked: '&?',
+    resetClicked: '&?',
+    cancelClicked: '&?',
+  },
```

Comment:
Step forward, but until the component stop doing that by itself by default, and stops taking all those extra inputs just to able to do this, not good enough..


One more attempt at explaining:

> We get into situations where data is changing and we don't know why. The change could be made by some code we wrote last year, four scopes down, but there is nothing in the code that we're looking at to remind us about it.
> It helps if we already have Isolated Components, because then we see who has access to the data that's changing...

(from https://teropa.info/blog/2015/10/18/refactoring-angular-apps-to-components.html, I recommend reading the whole thing before doing **any** component work)